public/app/features/alerting/unified/hooks/useExternalAmSelector.ts

```diff
@@ -1,76 +1,119 @@
+      refetchOnReconnect: true,
+      refetchOnFocus: true,
+    });
+
+  useEffect(() => {
+    fetchDataSourceSettings();
+    fetchActiveAlertmanagers();
+  }, [fetchActiveAlertmanagers, fetchDataSourceSettings]);
+
+  if (!alertmanagerDataSources) {
+    return [];
+  }
 
-  const droppedAMUrls = countBy(discoveredAlertmanagers?.droppedAlertManagers, (x) => x.url);
-  const activeAMUrls = countBy(discoveredAlertmanagers?.activeAlertManagers, (x) => x.url);
+  return alertmanagerDataSources.map<ExternalAlertmanagerDataSourceWithStatus>((dataSourceSettings) => {
+    const status = externalAlertmanagers
+      ? determineAlertmanagerConnectionStatus(externalAlertmanagers, dataSourceSettings)
+      : 'unknown';
 
-  return externalDsAlertManagers.map<ExternalDataSourceAM>((dsAm) => {
-    const dsSettings = alertmanagerDatasources[dsAm.uid];
+    return {
+      dataSourceSettings,
+      status,
+    };
+  });
+}
 
-    if (!dsSettings) {
-      return {
-        dataSource: dsAm,
-        status: 'pending',
-      };
-    }
+// using the information from /api/v1/ngalert/alertmanagers we should derive the connection status of a single data source
+function determineAlertmanagerConnectionStatus(
+  externalAlertmanagers: ExternalAlertmanagers,
+  dataSourceSettings: DataSourceSettings<AlertManagerDataSourceJsonData>
+): ConnectionStatus {
+  const isInterestedInAlerts = dataSourceSettings.jsonData.handleGrafanaManagedAlerts;
+  if (!isInterestedInAlerts) {
+    return 'uninterested';
+  }
 
-    const amUrl = getDataSourceUrlWithProtocol(dsSettings);
-    const amStatusUrl = `${amUrl}/api/v2/alerts`;
+  const isActive =
+    externalAlertmanagers?.activeAlertManagers.filter((am) => {
+      return isAlertmanagerMatchByURL(dataSourceSettings.url, am.url);
+    }) ?? [];
 
-    const matchingDroppedUrls = droppedAMUrls[amStatusUrl] ?? 0;
-    const matchingActiveUrls = activeAMUrls[amStatusUrl] ?? 0;
+  const isDropped =
+    externalAlertmanagers?.droppedAlertManagers.filter((am) => {
+      return isAlertmanagerMatchByURL(dataSourceSettings.url, am.url);
+    }) ?? [];
 
-    const isDropped = matchingDroppedUrls > 0;
-    const isActive = matchingActiveUrls > 0;
+  const isPending = isActive.length === 0 && isDropped.length === 0;
 
+  let status: ConnectionStatus = 'unknown';
+  if (isActive?.length === 1) {
+    status = 'active';
+  } else if (isDropped.length === 1) {
+    status = 'dropped';
+  } else if (isActive.length > 1) {
     // Multiple Alertmanagers of the same URL may exist (e.g. with different credentials)
     // Alertmanager response only contains URLs, so in case of duplication, we are not able
     // to distinguish which is which, resulting in an inconclusive status.
-    const isStatusInconclusive = matchingDroppedUrls + matchingActiveUrls > 1;
-
-    const status = isDropped ? 'dropped' : isActive ? 'active' : 'pending';
+    status = 'inconclusive';
```

Comment:
@konrad147 I've flattened `inconclusive` to a single state, I think before this refactor we could have both `active` and `inconclusive` at the same time? Do you think we should keep that behaviour?

Reply:
I think we should keep without flattening. It's possible to have inconclusive and active at the same time:

<img width="2114" alt="Screenshot 2024-01-23 at 17 14 30" src="https://github.com/grafana/grafana/assets/33540275/dbf7ae5f-3432-4841-ac1f-5b886d6fbcc6">

Reply:
This is a tough one – would the user care about what we _think_ the state is (based on another data source with the same URL but different credentials) if the likelihood of reporting the wrong status is high? :D

To illustrate imagine the following scenario:

```
AM1: http://url.com
AM2: http://url.com (with different credentials)
```

The internal AM responds with

```
active: ["http://url.com"],
dropped: ["http://url.com"]
```

This scenario describes a state where we would report "active" and "inconclusive" when there's a 50% chance we're wrong about it 😅

Reply:
Good question. Previously `inconclusive` was used if we found two or more AMs with the same URL. It was the only condition. 
The case I'm considering based on your example is what if the `dropped` array is empty? Then we have 2 AM datasources with the same URL but only one reported as `active`

Reply:
That is true, in that case both Alertmanagers will be `inconclusive` since one of them will be pending and we don't know which one :D